Index out of bound on containers with short names

## Bug Report

I've spawned kuberarmor on a kubernetes cluster, but noticed one pod constantly dying with the error:
```
panic: runtime error: slice bounds out of range 
```
After some hours of despair while debugging I found out that a container that we spawned on the machine manually called "htop" caused that. 
https://github.com/kubearmor/KubeArmor/blob/main/KubeArmor/core/containerdHandler.go#L135 causes kubearmor to fail on containers with a name/id less then 12 chars. Not sure if this is intended by design, but it should probably not panic because of this.

**General Information**

- k3s, containered
- Kernel 5.15.77
- kubernetes 1.25


**To Reproduce**

1. create containers with fixed Id's under 12 chars. e.g `ctr run --rm -d docker.io/library/debian:latest cont1`
2. run kubearmor on kubernetes
3. see logs for the node running the created container

**Expected behavior**

Not to fail on a `slice out of bounds` or at least not to crash the entire KubeArmor pod.

Morning. So I reinstalled Kubearmor this morning on our cluster and started a couple of htop containers on several hosts.
5 of 6 KubeArmor pods are failing with a output like this:

```
2023-03-27 08:05:38.868259	INFO	Build Time: 2023-03-26 15:03:06.511515418 +0000 UTC
2023-03-27 08:05:38.868403	INFO	Arguments [cluster:default coverageTest:false criSocket: defaultCapabilitiesPosture:audit defaultFilePosture:audit defaultNetworkPosture:audit enableKubeArmorHostPolicy:false enableKubeArmorPolicy:true enableKubeArmorVm:false gRPC:32767 host:bm-xxx hostDefaultCapabilitiesPosture:audit hostDefaultFilePosture:audit hostDefaultNetworkPosture:audit hostVisibility:default k8s:true kubeconfig: logPath:none lsm:bpf,apparmor,selinux seLinuxProfileDir:/tmp/kubearmor.selinux visibility:process,file,network,capabilities]
2023-03-27 08:05:38.868498	INFO	Configuration [{Cluster:default Host:bm-xxx GRPC:32767 LogPath:/tmp/kubearmor.log SELinuxProfileDir:/tmp/kubearmor.selinux CRISocket: Visibility:process,file,network,capabilities HostVisibility:default Policy:true HostPolicy:true KVMAgent:false K8sEnv:true DefaultFilePosture:audit DefaultNetworkPosture:audit DefaultCapabilitiesPosture:audit HostDefaultFilePosture:audit HostDefaultNetworkPosture:audit HostDefaultCapabilitiesPosture:audit CoverageTest:false LsmOrder:[]}]
2023-03-27 08:05:38.868529	INFO	Final Configuration [{Cluster:default Host:bm-xxx GRPC:32767 LogPath:/tmp/kubearmor.log SELinuxProfileDir:/tmp/kubearmor.selinux CRISocket: Visibility:process,file,network,capabilities HostVisibility:none Policy:true HostPolicy:true KVMAgent:false K8sEnv:true DefaultFilePosture:audit DefaultNetworkPosture:audit DefaultCapabilitiesPosture:audit HostDefaultFilePosture:audit HostDefaultNetworkPosture:audit HostDefaultCapabilitiesPosture:audit CoverageTest:false LsmOrder:[bpf apparmor selinux]}]
2023-03-27 08:05:38.868842	INFO	Initialized Kubernetes client
2023-03-27 08:05:38.868878	INFO	Started to monitor node events
2023-03-27 08:05:38.868922	INFO	GlobalCfg.Host=bm-xxx, KUBEARMOR_NODENAME=
2023-03-27 08:05:39.869385	INFO	Node Name: bm-xxx
2023-03-27 08:05:39.869507	INFO	Node IP: xxx.xxx.xxx.xxx
2023-03-27 08:05:39.869555	INFO	Node Annotations: map[alpha.kubernetes.io/provided-node-ip:xxx.xxx.xxx.xxx csi.volume.kubernetes.io/nodeid:{"driver.longhorn.io":"bm-xxx.xxx.xxx.xxx","rook-ceph.cephfs.csi.ceph.com":"bm-xxx.xxx.xxx.xxx","rook-ceph.rbd.csi.ceph.com":"bm-xxx.xxx.xxx.xxx","ru.yandex.s3.csi":"bm-xxx.xxx.xxx.xxx"} io.cilium.network.ipv4-cilium-host:10.0.0.228 io.cilium.network.ipv4-health-ip:10.0.0.247 io.cilium.network.ipv4-pod-cidr:10.0.0.0/24 k3s.io/node-args:["agent","--server","https://cp.k8s.<domain>.de:443","--token","********","--node-label","worker=metal","--node-ip","xxx.xxx.xxx.xxx","--node-external-ip","xxx.xxx.xxx.xxx","--kubelet-arg","cloud-provider=external"] k3s.io/node-config-hash:NZZFVEOEF3MUZUDKQA6SLF3UVW4H6WY4MS7NEGT7V2XT4MBVJWOA==== k3s.io/node-env:{"K3S_DATA_DIR":"/var/lib/rancher/k3s/data/691f04419e1d8a92fa96d43d6c1ad05162cda6e445ba30bc5eac758bf7597063"} kubearmor-policy:audited kubearmor-visibility:none kubevirt.io/heartbeat:2023-03-27T08:04:58Z node.alpha.kubernetes.io/ttl:0 volumes.kubernetes.io/controller-managed-attach-detach:true]
2023-03-27 08:05:39.869571	INFO	OS Image: Flatcar Container Linux by Kinvolk 3374.2.1 (Oklo)
2023-03-27 08:05:39.869585	INFO	Kernel Version: 5.15.77-flatcar
2023-03-27 08:05:39.869598	INFO	Kubelet Version: v1.25.5+k3s1
2023-03-27 08:05:39.869612	INFO	Container Runtime: containerd://1.6.12-k3s1
2023-03-27 08:05:39.870230	INFO	Initialized KubeArmor Logger
2023-03-27 08:05:39.871563	INFO	Initializing eBPF system monitor
2023-03-27 08:05:39.871594	INFO	eBPF system monitor object file path: /opt/kubearmor/BPF/system_monitor.bpf.o
2023-03-27 08:05:42.270924	INFO	Initialized the eBPF system monitor
2023-03-27 08:05:43.834434	INFO	Ignoring kprobe security_path_unlink
2023-03-27 08:05:43.834501	INFO	Ignoring kprobe security_path_rmdir
2023-03-27 08:05:43.927154	INFO	Initialized KubeArmor Monitor
2023-03-27 08:05:43.927202	INFO	Started to monitor system events
2023-03-27 08:05:43.929364	INFO	Supported LSMs: lockdown,capability,selinux
2023-03-27 08:05:43.929389	INFO	Disabled KubeArmor Enforcer since No LSM is enabled
2023-03-27 08:05:43.929403	INFO	CRI socket not set. Trying to detect.
2023-03-27 08:05:43.929758	INFO	Initialized Containerd Handler
2023-03-27 08:05:43.929775	INFO	Started to monitor Containerd events
2023-03-27 08:05:43.974281	INFO	Detected a container (added/5fea88d2bfb3)
2023-03-27 08:05:43.999677	INFO	Detected a container (added/35902153a33b)
2023-03-27 08:05:44.024335	INFO	Detected a container (added/12bc8288757e)
2023-03-27 08:05:44.045651	INFO	Detected a container (added/05518d00e2fe)
2023-03-27 08:05:44.071689	INFO	Detected a container (added/f087b6323e12)
2023-03-27 08:05:44.096769	INFO	Detected a container (added/a5783044bf77)
2023-03-27 08:05:44.120641	INFO	Detected a container (added/5a9976c437fd)
2023-03-27 08:05:44.144626	INFO	Detected a container (added/d26f69e0f15b)
2023-03-27 08:05:44.165887	INFO	Detected a container (added/dbcf9f94bf33)
2023-03-27 08:05:44.188816	INFO	Detected a container (added/89666bc1b94d)
2023-03-27 08:05:44.209270	INFO	Detected a container (added/c47fe8e9e2df)
2023-03-27 08:05:44.238758	INFO	Detected a container (added/450291057edc)
2023-03-27 08:05:44.249765	INFO	Detected a container (added/fd83d531bf25)
2023-03-27 08:05:44.270937	INFO	Detected a container (added/ff8cf7487cfd)
2023-03-27 08:05:44.292992	INFO	Detected a container (added/279479589528)
2023-03-27 08:05:44.302627	INFO	Detected a container (added/04a063f88690)
2023-03-27 08:05:44.322713	INFO	Detected a container (added/6bfde2bb5a52)
panic: runtime error: slice bounds out of range [:12] with length 4

goroutine 111 [running]:
github.com/kubearmor/KubeArmor/KubeArmor/core.(*ContainerdHandler).GetContainerInfo(0xc004c1a360, {0x1f90820, 0xc000876600}, {0xc0019da7cc, 0x4})
	/usr/src/KubeArmor/KubeArmor/core/containerdHandler.go:135 +0x9c5
github.com/kubearmor/KubeArmor/KubeArmor/core.(*KubeArmorDaemon).UpdateContainerdContainer(0xc00054a000, {0x1f90820?, 0xc000876600?}, {0xc0019da7cc, 0x4}, {0x1cc4e06?, 0x0?})
	/usr/src/KubeArmor/KubeArmor/core/containerdHandler.go:258 +0xce
github.com/kubearmor/KubeArmor/KubeArmor/core.(*KubeArmorDaemon).MonitorContainerdEvents(0xc00054a000)
	/usr/src/KubeArmor/KubeArmor/core/containerdHandler.go:395 +0x5da
created by github.com/kubearmor/KubeArmor/KubeArmor/core.KubeArmor
	/usr/src/KubeArmor/KubeArmor/core/kubeArmor.go:557 +0x13ad

```
Screenshot of the pods:
![Screen Shot 2023-03-27 at 10 28 54](https://user-images.githubusercontent.com/1640534/227885740-3fdd345a-4fcc-48e4-b7b4-c769a60bccd0.png)

Below is the list of running containers for the same node. And this is where my assumption of htop being the culprit emerged. it's the only one reported by ctr having a name/id less than 12 chars. If I delete the htop container with ctr kubearmor will work just fine.


```
CONTAINER                                                           IMAGE                                                                                            RUNTIME
01ce38331b8b5ad013d4dbe3861b552505e576216a22c34a639594cbc34e66f7    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
02ccc3fd2260d40d7d5cbcfb42096208e33d4c935daec6524bf02958c39b2dc5    ghcr.io/kubereboot/kured:1.12.0                                                                  io.containerd.runc.v2
04a063f886906805f5c08941f84a7c47dcdf19b49537517b2d748203d9772484    docker.io/nginx/nginx-ingress:3.0.1                                                              io.containerd.runc.v2
05518d00e2fe6aba60874641dcb85cd3ab56ede0242fad366fb2a4c066662864    quay.io/k8scsi/csi-node-driver-registrar:v1.2.0                                                  io.containerd.runc.v2
0670db44632cb417f8af7b27f671f5067a7df855660d783070a0da2bc855dc80    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
07c78c76e182bb57d171a484dbfe72b7247ae3292042f96b2c0bd3228d4e14f5    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
0ec9d9859764fc6336d6419513e302f2072e7f9a70fa9aaa74ccf2c8b15c882c    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
11785e4709c0dbb4ce48960508292b781b242fae30c02d73f3e7a87a87565ce3    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
117f0881a75657b072aa9c421c696a2206dce3a1d884ea3503ebedb225add0f7    quay.io/cilium/cilium@sha256:454134506b0448c756398d3e8df68d474acde2a622ab58d0c7e8b272b5867d0d    io.containerd.runc.v2
12bc8288757e0486deb3b889750261ca29a58252e122b8d504d4aca8f4d64ba2    quay.io/cephcsi/cephcsi:v3.8.0                                                                   io.containerd.runc.v2
18a3742692a5fa4e7a767ce3d80b6c1c9cc8bf8442bfaa6c2c11f9fdc0904f42    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
1a470304c68987d5c8995a6eddd16a9e75aa4a47ec3e028ab1a013b5e3efb1f3    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
1ead251b4984ff6f0dc70ac2ae2203b0aef48b76bea859e39fe3665735c30fad    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
1fb6fd27003c2c78c6768588afd7b4a4358cfc0175d4474781f3281a804d48b2    quay.io/prometheus/node-exporter:v1.1.2                                                          io.containerd.runc.v2
279479589528cac125c7237df2a04543d20d3693ac343fef21e295a3f21868f6    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
2a8f07cc46e4e2d7ee917fc9282627156d78bca2dbde29f6e03696b27ab47157    docker.io/rook/ceph:v1.11.2                                                                      io.containerd.runc.v2
2b2c337c7f04f6675f0f61d1ba3b8d2c16a43bd9fd8ffea28a8530d803eef156    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
2e12215e6fd3b2edd1dc29d322b0b5401cf14631e3d26cce4c1cadc530c6defd    ghcr.io/ondat/trousseau:trousseau-d3e4f2569b2eddeea992e47dae29a931182379dd                       io.containerd.runc.v2
2eacb3d729bd7edf3e66b7aea5fac06c9bb91583c21e162f24779fefdb62196e    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
303439ad3419ad00aecc38085ffae6365503ade65362b9f5836958140897751b    ghcr.io/ondat/trousseau:vault-d3e4f2569b2eddeea992e47dae29a931182379dd                           io.containerd.runc.v2
30efddfcf0cdc5d9f640196bb41f5c7022a794c609f967f74862ce7cfd84a1dc    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
35902153a33bd690bd6f6720e51aa6e7719511263131746a404dd57da7c34354    docker.io/longhornio/longhorn-instance-manager:v1.4.0                                            io.containerd.runc.v2
35a9c37f897de901d05f813647465326eb311b5d62deed9c75e63188a0882948    docker.io/falcosecurity/falcoctl:0.4.0                                                           io.containerd.runc.v2
35e9c0ff5594a85aa83aa4417253c01bc69f05a89220d5e1009bf116c9be3dc2    registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.7.0                                     io.containerd.runc.v2
39089441a6d9f7ac9602324b2d28a0bc3c3d355c60ad8ea8093406ea47ab1005    docker.io/falcosecurity/falco-no-driver:0.34.1                                                   io.containerd.runc.v2
3a6f3ac72ad7389bc9c1a040da18c26a392a92ebeb27ece8fe99b664b6514c4b    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
3ef81b4f4f563d8fb6f3f21d05cc586b9ba343895e276656351621ef0e3d6d3d    registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.7.0                                     io.containerd.runc.v2
40b597640051341e38f78c5af345174c8af2a3e4b9a06656a89bbd2b5058dc9e    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
41090936c29ff09473b52bdefc492abfd3090b6fad9f011bcce6682b79b20796    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
41e859879487aef38640122c3a41a6d5b39e0fd1e956841648f1e8c9721878c8    ghcr.io/kubereboot/kured:1.12.0                                                                  io.containerd.runc.v2
44b300e942c73411102c52f42cdf575fa0125b15357f3d8d72701c9696d80e9a    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
450291057edc412b744d0934409d9218aa3e7c8803d3f853ad066fa8dbe1aaa7    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
482d12d360987994eea25f2e41010614d967e399082b54a0a8095625a35215e9    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
4c4cceb5122f94db0538278647692a0094de7f43b95135bd1f39c23106f27104    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
4e0f8136a572396e595222f2ad8f39087f9a1645d0d558943ad62d6d1ccf0c5b    quay.io/cephcsi/cephcsi:v3.8.0                                                                   io.containerd.runc.v2
4fac28d66de67d83178df114c84c842992f497559b78cd906b057a6f2b856de2    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
51781cfd0b5ab6294835b3d16cbcd17ff59c38048d8f4efcac5bd237032d3531    quay.io/cephcsi/cephcsi:v3.8.0                                                                   io.containerd.runc.v2
555c142419865f36bfdf2af69a0b4f652847877f18edb87d545e28fe8563e26c    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
569a877b9dfcec450d1876543d1ef5159101b9baf0a42317dc1ca400183da8fb    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
579960e0d328cd57717e73e3e34b6bd4c16b9e26fc5b7ed2bdcfa9691541a9b8    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
57a730146180d02af6b5e1dff2c1de8bb60bc73e34264a20540dd7ec79708a07    cr.yandex/crp9ftr22d26age3hulg/csi-s3:0.34.2                                                     io.containerd.runc.v2
5a9976c437fd1a2bd5348f8b0135be6c9b83b8dd835b6ce7fa2b8956a460ff83    quay.io/cilium/cilium@sha256:454134506b0448c756398d3e8df68d474acde2a622ab58d0c7e8b272b5867d0d    io.containerd.runc.v2
5e9c7b69f27d57a22563584c9652e7ecdd39a2b6d2b741115750367ee2af7c3e    quay.io/cephcsi/cephcsi:v3.8.0                                                                   io.containerd.runc.v2
5fea88d2bfb3e3f97cab0a72878358f13a620d1c99e0ea3a28e1d83a28154843    quay.io/kubevirt/virt-handler:v0.58.0                                                            io.containerd.runc.v2
606aeaa36ca1e34e1cdf85c1057d9d5ab276ab02fc09308c8f706e125cb4bff1    quay.io/k8scsi/csi-node-driver-registrar:v1.2.0                                                  io.containerd.runc.v2
62438498afcf1d97fa270094e00549bcf3e94158462b26d3fc2f41758aae3bee    quay.io/cephcsi/cephcsi:v3.8.0                                                                   io.containerd.runc.v2
62fb59de7d26c75b715d9212d52c44d93264f95c5bb06876e9318f5da09ee939    ghcr.io/ondat/trousseau:awskms-d3e4f2569b2eddeea992e47dae29a931182379dd                          io.containerd.runc.v2
637f92f9c4b7e2dbc1cecc6d3fb618b1abadabc94623ec6dfa16851573601473    ghcr.io/ondat/trousseau:trousseau-d3e4f2569b2eddeea992e47dae29a931182379dd                       io.containerd.runc.v2
65bb65c1f48d07005c99642cbb29a68370b71e00aa45f4ba0d6daee3592f9a5e    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
67130a585207df53be5c75cd6863a653d8c6c79b1641baa3ec7a79a1dadeaf29    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
67579391529ac2b1307489596ca1ea3cbbe590f76aae6f1087463f6236aaa54c    docker.io/library/busybox:stable-glibc                                                           io.containerd.runc.v2
67e04d37f2d19cf1624b316be4afe72838c9adf81e34608faf79b33b62220964    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
681c1bb9f99e303a7d130b09e154b9f5aa8e7731b7b0a6fef3c283751e28851b    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
6a7aec951f74b8e0926dd62f1cad045034b2c1738048e282a59a877ac18469fa    docker.io/longhornio/longhorn-manager:v1.4.0                                                     io.containerd.runc.v2
6bc420c2b9828841cf3f524400bb5c7c5b85ff5c620386dcd4157a93029707c2    docker.io/longhornio/longhorn-manager:v1.4.0                                                     io.containerd.runc.v2
6bfde2bb5a52f76463a1550e7438a0fa91ed8322775a03960bfa8d8e82f183b9    docker.io/longhornio/livenessprobe:v2.8.0                                                        io.containerd.runc.v2
75b47737ab5474803691265f60ad1a89d567b669d51961e65eda3db0de1ad632    docker.io/loftsh/vcluster:0.13.0                                                                 io.containerd.runc.v2
793e0f762d8e1b5b815e5f02f3589ad7b6c12044d454d4e308bc50b94188aa04    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
7afc2fb07c795b71c883c122882fe3091fa4667d63829ddfde9799a6551bc42d    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
7e347182e79a63d8c5b3523a23224eb212dae084ee99b2055e9d9692481c0f88    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
7effe72150641c2d94eb153c3aa539050a20689f69cfd4d432211ef0d0d25ab4    ghcr.io/ondat/trousseau:vault-d3e4f2569b2eddeea992e47dae29a931182379dd                           io.containerd.runc.v2
8179de709acac44a7a237df3e04e083640f3326ca35ca147fb533376fd3e1275    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
8372d1cbd74c44b69309968d948021682a89257cccc410d94bbf493535fdf2a2    docker.io/library/busybox:stable-glibc                                                           io.containerd.runc.v2
860439323ac8f0c0fc473a5e7fc2478439176db71da99e952b2fd096b8f88acb    registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.7.0                                     io.containerd.runc.v2
86ba0a4819f7209bf4f9405c2b3e2c572b53e9627bb3229c9b8f51297b8f1af3    cr.yandex/crp9ftr22d26age3hulg/csi-s3:0.34.2                                                     io.containerd.runc.v2
88a2e5363b39a8767a2c7828605c38aef2e7d0781f7c51db3f8bf747e6315f73    ghcr.io/ondat/trousseau:awskms-d3e4f2569b2eddeea992e47dae29a931182379dd                          io.containerd.runc.v2
88bb273d2288488953f86bd5a5f54c495c797fcc622123b282929d2732dfd0f6    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
88bed4e3a4ed8de64962ee54cd7ab0a067d89d5dcf9156cea91cb00d7a0ffe83    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
89666bc1b94dcbaa591945439fbc98d2ddd5738013b554bfdc28509682607f2c    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
897cf6d5ef3b22d3006e4aa0e61860a55531cb023be3dc68a560bd406a8095ad    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
8f3fb7e2bbdb422d13fbfcc82a834feb976e20ab23cc8816abffcd6e149ec96f    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
90f87895a011807d6472f025c97e4bafe76ad2be24b36ffb4ecb6a20d894d9a9    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
9204327fa79b08dd34e1a5dff6a4e704bdca8b0f32847988ad099088062479b7    docker.io/longhornio/longhorn-instance-manager:v1.4.0                                            io.containerd.runc.v2
968dd0b533f85f178fa36ea9e5e5a6189540da1276ce990e2fbd398c0cdf9446    docker.io/falcosecurity/falco-no-driver:0.34.1                                                   io.containerd.runc.v2
990d066df43ed160246bcb73dca3d95aa11c1a77bbde0a2805246a99f088f5b3    docker.io/library/busybox:stable-glibc                                                           io.containerd.runc.v2
99f0a3f201db070de6d92387658c2f657ea9f85b4c8118549cbe26e61c15a6d9    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
9d55490edb078a95e72610cbbe58ccb86994081ec7ced7d4e04e92b37b5270f8    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
9e75c4e105b0ec5cf5129caacc79051af2471d1c22ef3b47aec21a29da934b51    docker.io/falcosecurity/falcoctl:0.4.0                                                           io.containerd.runc.v2
a06614dcbeaba32f53deb42babbc75dd54a98895db00e1c53237ba0de4d6caed    docker.io/kubearmor/kubearmor:v0.8.0                                                             io.containerd.runc.v2
a1377d468c66e88410e212fe4595924a596ea5bec058588bf4a2aafcf5881170    quay.io/kubevirt/virt-handler:v0.58.0                                                            io.containerd.runc.v2
a30bf67020e6918da1ad78e6e56e2ab7c807f61ec74cd721dc0255427cc34738    docker.io/longhornio/livenessprobe:v2.8.0                                                        io.containerd.runc.v2
a31d76e471f8be45108398e4d3343b99cb77b18497eba253dc28ccfd216e745f    docker.io/rook/ceph:v1.11.2                                                                      io.containerd.runc.v2
a4bd947e95ed5168e91deb89b2e883e45fe23d200e7959a784a19866e1130ebe    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
a5783044bf771afe4f551bc515398eec9b1871c74cbc7d54b584a4892808689e    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
a67e1b166e61cb8cdf93fb4409064cee4cbad428db0eddf4a22177eaea52d740    docker.io/kubearmor/kubearmor-init:v0.8.0                                                        io.containerd.runc.v2
a7fd63716071128d99d3b8c0aa494dc1f88a7c884f1942797700ad3d4e8bab25    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
a9be314b42b131f8aee92c79883abc84066cc9777f5b99247ff13c1e14088602    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
aa44acbeb93ba6708fe7dd1b697eb78f4aae64f3e0ed0888d04e114f5a1b8add    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
ab15ab4659dfa89acb70750bc8a3857fa8968948f9c2ca738ca6a07068c581c9    docker.io/longhornio/longhorn-engine:v1.4.0                                                      io.containerd.runc.v2
b012c1eface585b6854b0b307a2550c5c5be0e43c9a99a767772d63d79185f21    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
b6bbc436e5bef0d1f7566bf395e5a7e4e38d012aaa46beb18118a3ffec467ca9    docker.io/rook/ceph:v1.11.2                                                                      io.containerd.runc.v2
b799f65ced5bdeba461b2f22857a17263c6a8d8fe04110b0fbd9f7e742651a7d    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
b892c62f240031d73aa6bca05ba9091df746e15c391b81ecc71bc870ca27dd64    docker.io/falcosecurity/falcoctl:0.4.0                                                           io.containerd.runc.v2
b8ed477c6e4c9be62fb733e7377d87054265f018de484eb0735736d4a6704cb3    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
b9250c941a60b4d4fd6afee891ab6c3d2247dc3086799bec991412bf388b29aa    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
bebd44d675a32c98beb1f2d9b8a7d4bf0dfc9067eeda433e3772d9b731388ebb    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
bf18ebee67aeddd65b6652767138d8e865f43a9ad089eb743f94c4471dae097a    docker.io/longhornio/longhorn-manager:v1.4.0                                                     io.containerd.runc.v2
bf36383541745faae63a83bacd6f7cdde37ace982a65a4b394796b91224f02f5    docker.io/nginx/nginx-ingress:3.0.1                                                              io.containerd.runc.v2
c08f7477595514520aa16373006bf34f67c5e3f401b94b530d298fd1b5fe004d    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
c1ada807abb79c55cd1d945f176de498393be2a7f98c3fcd98577889b02a9fde    docker.io/longhornio/longhorn-manager:v1.4.0                                                     io.containerd.runc.v2
c1afeeb05a196d8df2ab2c23e3c638e4b2c30646af646f3b29dbd1fbcd5cfd8f    quay.io/cilium/cilium@sha256:454134506b0448c756398d3e8df68d474acde2a622ab58d0c7e8b272b5867d0d    io.containerd.runc.v2
c335f8741028c07366d0f3a9ced71f4876e145ee2bc45df386f8cedb525ff5b5    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
c47fe8e9e2df156630573d851dee135d7cd5c48b5133b7932ed6bc07d9b0ffc7    quay.io/prometheus/node-exporter:v1.1.2                                                          io.containerd.runc.v2
c53e40c79e1bffde67bd6057ff0b177c37f4019a4a8e960d30128a6980cd1ad7    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
c650b838d9857150e64d3a6ff8e5b0cfee4f8cbcfdadc0e9bb4f1918c3e53ce2    quay.io/kubevirt/virt-launcher:v0.58.0                                                           io.containerd.runc.v2
c6bdd9c23d3397a8ba471314e36926a1bdb6c27d1b615195ad416d1722a3a52b    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
c72be9db0a1683e76d25a1a7d781e488872e30942a467506b955d3d3917af726    quay.io/cephcsi/cephcsi:v3.8.0                                                                   io.containerd.runc.v2
c9eaf3c4b61d653b0543fa8aaa0f158584b8d7c02194448a4b023bdf0f3b68f5    quay.io/cilium/cilium@sha256:454134506b0448c756398d3e8df68d474acde2a622ab58d0c7e8b272b5867d0d    io.containerd.runc.v2
d26f69e0f15b71d0e1bff7ab2cec044976ad975d333845657e017821ae7359af    docker.io/longhornio/longhorn-engine:v1.4.0                                                      io.containerd.runc.v2
d27333d8752de84fbcea020a6be204c3ecfee6dc612efd72331b4a882c768e4f    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
d59ca72d99dfd8ab5204390b97103844a45bbb134357cd84b9922cc2c0acae1b    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
d84531d39b9668b770d21e931a97e92c3e1f2cf2c54200d98d6eb3fb76ac24d8    docker.io/longhornio/csi-node-driver-registrar:v2.5.0                                            io.containerd.runc.v2
d92c81e751bd3a23dbf61dcddccd42158a3a53e58509ad29c2e6c7965c7b6d81    quay.io/cephcsi/cephcsi:v3.8.0                                                                   io.containerd.runc.v2
da32190ed0214745d546749c8fc7adc8edf684984ee37fb984c2d184df70e019    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
dafa91de8190afc35ac9523dc6555e48fb4ecdf5a46c8f3a26c2e65d4636efd9    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
dbcf9f94bf331d247389c2742c1051e15314244d1dbe6aee901b3aa24741832a    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
dc5e6fd4f249718b55e35d61255c16940ad6d9ac479fbedc8e001429134c9aa0    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
dd2d8def60aaf7551bfed25c103ee10243e02872666be82bc81435b18460160f    quay.io/cephcsi/cephcsi:v3.8.0                                                                   io.containerd.runc.v2
e37c4f7bc9312ea004da4822a9561478f9931d462c90dfe75a05b72ed55aa24a    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
e91b5836be89c4042f6a4391dc933ac296a30842c185692eeb1533c7160194d9    quay.io/cilium/cilium@sha256:454134506b0448c756398d3e8df68d474acde2a622ab58d0c7e8b272b5867d0d    io.containerd.runc.v2
ed731adc3c8332c808daf0ce02f9ae0139efde280e6c1e284da97779183d19ea    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
f087b6323e12083fa5cc2b70ccfdb1a8849421e330bbdc9435a396e64b1243ba    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
f2530b2de0b587e2b39fccdc5df98ddfd96cb7169cfd411f86eb062f9b61167b    docker.io/falcosecurity/falco-driver-loader:0.34.1                                               io.containerd.runc.v2
f26fe6ddd47c611bbbc1faf4d110976331a45eb85d25c202fe008b73ae8e0db0    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
f37d97339cf9333c581633316f571a00e05a5209caf7d6c9a471adcdb6fa81a1    docker.io/longhornio/longhorn-manager:v1.4.0                                                     io.containerd.runc.v2
f4417a5bfdb271bed6a6c190afb02aef9700be7b92ba508961e71052e943b5a0    ghcr.io/ondat/trousseau:proxy-d3e4f2569b2eddeea992e47dae29a931182379dd                           io.containerd.runc.v2
f6b7726097aba1fd96974bf37840c643b456f44ccfff1d86d5fd6ea956dc33d8    quay.io/cilium/cilium@sha256:454134506b0448c756398d3e8df68d474acde2a622ab58d0c7e8b272b5867d0d    io.containerd.runc.v2
f7bc303b9efecfed29d00e4e64e8759671af0befc8d0217cdc6aa333394f1e10    docker.io/longhornio/csi-node-driver-registrar:v2.5.0                                            io.containerd.runc.v2
f94f5097dbcd3929ad677ecd117632255a35461d132c155b3294a306def98f1e    registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.7.0                                     io.containerd.runc.v2
fbfc20d6d3f005a0db9cafac5c039c089114c9b1ac2acec8957deb5bb2a7c8c5    docker.io/rancher/mirrored-pause:3.6                                                             io.containerd.runc.v2
fd83d531bf25ab2d30af4f0d1be4e834a8dc33a54a9a1183500ee54409168c2c    quay.io/ceph/ceph:v17.2.5                                                                        io.containerd.runc.v2
fe9839692e555886c899d3881c48eb751e16ce68b7ba155d575cf78cfe2deff5    ghcr.io/ondat/trousseau:proxy-d3e4f2569b2eddeea992e47dae29a931182379dd                           io.containerd.runc.v2
ff8cf7487cfd2942aeac6aec3c8f73e227de758f9c5971bf9d1e925a40169b93    docker.io/loftsh/vcluster:0.13.0                                                                 io.containerd.runc.v2
htop                                                                <private-domain>/htop:3.2.1                                                                      io.containerd.runc.v2
```

--- a/KubeArmor/core/containerdHandler.go
+++ b/KubeArmor/core/containerdHandler.go
@@ -132,8 +132,11 @@ func (ch *ContainerdHandler) GetContainerInfo(ctx context.Context, containerID s
 	// == container base == //
 
 	container.ContainerID = res.Container.ID
-	container.ContainerName = res.Container.ID[:12]
-
+	if len(res.Container.ID) < 12 {
+		container.ContainerName = res.Container.ID
+	} else {
+		container.ContainerName = res.Container.ID[:12]
+	}
 	container.NamespaceName = "Unknown"
 	container.EndPointName = "Unknown"
 
@@ -318,7 +321,7 @@ func (dm *KubeArmorDaemon) UpdateContainerdContainer(ctx context.Context, contai
 			dm.RuntimeEnforcer.RegisterContainer(containerID, container.PidNS, container.MntNS)
 		}
 
-		dm.Logger.Printf("Detected a container (added/%s/pidns=%d/mntns=%d)", containerID[:12], container.PidNS, container.MntNS)
+		dm.Logger.Printf("Detected a container (added/%s/pidns=%d/mntns=%d)", container.ContainerName, container.PidNS, container.MntNS)
 
 	} else if action == "destroy" {
 		dm.ContainersLock.Lock()
@@ -353,7 +356,7 @@ func (dm *KubeArmorDaemon) UpdateContainerdContainer(ctx context.Context, contai
 			dm.RuntimeEnforcer.UnregisterContainer(containerID)
 		}
 
-		dm.Logger.Printf("Detected a container (removed/%s/pidns=%d/mntns=%d)", containerID[:12], container.PidNS, container.MntNS)
+		dm.Logger.Printf("Detected a container (removed/%s/pidns=%d/mntns=%d)", container.ContainerName, container.PidNS, container.MntNS)
 	}
 
 	return true

--- a/KubeArmor/core/containerdHandler.go
+++ b/KubeArmor/core/containerdHandler.go
@@ -132,11 +132,7 @@ func (ch *ContainerdHandler) GetContainerInfo(ctx context.Context, containerID s
 	// == container base == //
 
 	container.ContainerID = res.Container.ID
-	if len(res.Container.ID) < 12 {
-		container.ContainerName = res.Container.ID
-	} else {
-		container.ContainerName = res.Container.ID[:12]
-	}
+	container.ContainerName = res.Container.ID
 	container.NamespaceName = "Unknown"
 	container.EndPointName = "Unknown"
 
@@ -321,7 +317,7 @@ func (dm *KubeArmorDaemon) UpdateContainerdContainer(ctx context.Context, contai
 			dm.RuntimeEnforcer.RegisterContainer(containerID, container.PidNS, container.MntNS)
 		}
 
-		dm.Logger.Printf("Detected a container (added/%s/pidns=%d/mntns=%d)", container.ContainerName, container.PidNS, container.MntNS)
+		dm.Logger.Printf("Detected a container (added/%.12s/pidns=%d/mntns=%d)", containerID, container.PidNS, container.MntNS)
 
 	} else if action == "destroy" {
 		dm.ContainersLock.Lock()
@@ -356,7 +352,7 @@ func (dm *KubeArmorDaemon) UpdateContainerdContainer(ctx context.Context, contai
 			dm.RuntimeEnforcer.UnregisterContainer(containerID)
 		}
 
-		dm.Logger.Printf("Detected a container (removed/%s/pidns=%d/mntns=%d)", container.ContainerName, container.PidNS, container.MntNS)
+		dm.Logger.Printf("Detected a container (removed/%.12s/pidns=%d/mntns=%d)", containerID, container.PidNS, container.MntNS)
 	}
 
 	return true

--- a/KubeArmor/core/crioHandler.go
+++ b/KubeArmor/core/crioHandler.go
@@ -258,7 +258,7 @@ func (dm *KubeArmorDaemon) UpdateCrioContainer(ctx context.Context, containerID,
 			dm.RuntimeEnforcer.RegisterContainer(containerID, container.PidNS, container.MntNS)
 		}
 
-		dm.Logger.Printf("Detected a container (added/%s)", containerID[:12])
+		dm.Logger.Printf("Detected a container (added/%s)", container)
 	} else if action == "destroy" {
 		dm.ContainersLock.Lock()
 		container, ok := dm.Containers[containerID]
@@ -292,7 +292,7 @@ func (dm *KubeArmorDaemon) UpdateCrioContainer(ctx context.Context, containerID,
 			dm.RuntimeEnforcer.UnregisterContainer(containerID)
 		}
 
-		dm.Logger.Printf("Detected a container (removed/%s)", containerID[:12])
+		dm.Logger.Printf("Detected a container (removed/%s)", container.ContainerName)
 	}
 
 	return true

--- a/KubeArmor/core/crioHandler.go
+++ b/KubeArmor/core/crioHandler.go
@@ -258,7 +258,7 @@ func (dm *KubeArmorDaemon) UpdateCrioContainer(ctx context.Context, containerID,
 			dm.RuntimeEnforcer.RegisterContainer(containerID, container.PidNS, container.MntNS)
 		}
 
-		dm.Logger.Printf("Detected a container (added/%s)", container)
+		dm.Logger.Printf("Detected a container (added/%.12s)", containerID)
 	} else if action == "destroy" {
 		dm.ContainersLock.Lock()
 		container, ok := dm.Containers[containerID]
@@ -292,7 +292,7 @@ func (dm *KubeArmorDaemon) UpdateCrioContainer(ctx context.Context, containerID,
 			dm.RuntimeEnforcer.UnregisterContainer(containerID)
 		}
 
-		dm.Logger.Printf("Detected a container (removed/%s)", container.ContainerName)
+		dm.Logger.Printf("Detected a container (removed/%.12s)", containerID)
 	}
 
 	return true

--- a/KubeArmor/core/dockerHandler.go
+++ b/KubeArmor/core/dockerHandler.go
@@ -274,7 +274,7 @@ func (dm *KubeArmorDaemon) GetAlreadyDeployedDockerContainers() {
 					dm.RuntimeEnforcer.RegisterContainer(container.ContainerID, container.PidNS, container.MntNS)
 				}
 
-				dm.Logger.Printf("Detected a container (added/%s)", container.ContainerName)
+				dm.Logger.Printf("Detected a container (added/%.12s)", container.ContainerID)
 			}
 		}
 	}
@@ -357,7 +357,7 @@ func (dm *KubeArmorDaemon) UpdateDockerContainer(containerID, action string) {
 			dm.RuntimeEnforcer.RegisterContainer(containerID, container.PidNS, container.MntNS)
 		}
 
-		dm.Logger.Printf("Detected a container (added/%s)", container.ContainerName)
+		dm.Logger.Printf("Detected a container (added/%.12s)", containerID)
 
 	} else if action == "stop" || action == "destroy" {
 		// case 1: kill -> die -> stop
@@ -396,7 +396,7 @@ func (dm *KubeArmorDaemon) UpdateDockerContainer(containerID, action string) {
 			dm.RuntimeEnforcer.UnregisterContainer(containerID)
 		}
 
-		dm.Logger.Printf("Detected a container (removed/%s)", container.ContainerName)
+		dm.Logger.Printf("Detected a container (removed/%.12s)", containerID)
 	}
 }
 

--- a/KubeArmor/core/dockerHandler.go
+++ b/KubeArmor/core/dockerHandler.go
@@ -274,7 +274,7 @@ func (dm *KubeArmorDaemon) GetAlreadyDeployedDockerContainers() {
 					dm.RuntimeEnforcer.RegisterContainer(container.ContainerID, container.PidNS, container.MntNS)
 				}
 
-				dm.Logger.Printf("Detected a container (added/%s)", container.ContainerID[:12])
+				dm.Logger.Printf("Detected a container (added/%s)", container.ContainerName)
 			}
 		}
 	}
@@ -357,7 +357,7 @@ func (dm *KubeArmorDaemon) UpdateDockerContainer(containerID, action string) {
 			dm.RuntimeEnforcer.RegisterContainer(containerID, container.PidNS, container.MntNS)
 		}
 
-		dm.Logger.Printf("Detected a container (added/%s)", containerID[:12])
+		dm.Logger.Printf("Detected a container (added/%s)", container.ContainerName)
 
 	} else if action == "stop" || action == "destroy" {
 		// case 1: kill -> die -> stop
@@ -396,7 +396,7 @@ func (dm *KubeArmorDaemon) UpdateDockerContainer(containerID, action string) {
 			dm.RuntimeEnforcer.UnregisterContainer(containerID)
 		}
 
-		dm.Logger.Printf("Detected a container (removed/%s)", containerID[:12])
+		dm.Logger.Printf("Detected a container (removed/%s)", container.ContainerName)
 	}
 }
 

